cmd/livepeer/livepeer.go

```diff
@@ -676,6 +678,17 @@ func main() {
 			server.AuthWebhookURL = *authWebhookURL
 		}
 
+		isLocalHTTP, err := isLocalURL("https://" + *httpAddr)
+		if err != nil {
+			glog.Errorf("Error checking for local -httpAddr: %v", err)
+			return
+		}
+		if !*httpIngest && !isLocalHTTP && server.AuthWebhookURL == "" {
```

Comment:
This is one place where default values are tricky. How can the user still disable HTTP ingest even with localhost and/or a webhook?

Does golang CLI offer a way to distinguish between "user set the flag" and "user did not set the flag, so using the default value" ?

Option types would work much better here, but golang doesn't really have a notion of that...

Reply:
Ah yeah meant to look into this further.

> Does golang CLI offer a way to distinguish between "user set the flag" and "user did not set the flag, so using the default value" ?

46db19e: Use [flag.Visit()](https://golang.org/pkg/flag/#Visit) to loop through all set flags to populate a map that tracks set flags.

f51a50b: Add a note in the HTTP ingest docs about always disabling HTTP ingest.

Unfortunately, it doesn't look like the `flag` package exposes a more convenient way to check if a flag is set ([cobra](https://github.com/spf13/cobra), another CLI package, exposes a `Changed()` method on flags that can be used to check if the flag was set).

Also changed the default value of `-httpIngest` to true and now the value is set to false if the flag is not set and a non-local `-httpAddr` is used without an auth webhook URL.

```golang
if !isFlagSet["httpIngest"] && !isLocalHTTP & server.AuthWebhookURL == "" {
    ...
    *httpIngest = false
}
```

Reply:
Fantastic. Thanks!